server/repository/src/migrations/v1_08_00/invoice_required_currency_id.rs

```diff
@@ -0,0 +1,18 @@
+use crate::StorageConnection;
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    use crate::migrations::sql;
+
+    sql!(
+        connection,
+        r#"
+            ALTER TABLE invoice RENAME COLUMN currency_id TO old_currency_id;
+            ALTER TABLE invoice ADD COLUMN currency_id TEXT REFERENCES currency(id) NOT NULL;
+            UPDATE invoice SET currency_id = old_currency_id;
+            UPDATE invoice SET currency_id = (SELECT id FROM currency WHERE is_home_currency = true) AND currency_id = '';
```

Comment:
This auto updates any old invoices and also any invoices synced from OG to have `is_home_currency` currency save by default so shouldn't cause problems with any updates from old versioning

Reply:
I thought so too and took me ages to get it working... it doesn't like the WHERE currency_id = ''